In [39]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer
from sklearn.decomposition import TruncatedSVD
from sklearn.cluster import KMeans
from scipy.spatial.distance import cosine
from sklearn.feature_selection import SelectKBest
from sklearn.metrics import calinski_harabasz_score
from sklearn.metrics import silhouette_score
from scipy.spatial.distance import euclidean, cityblock
import nltk
import seaborn as sns
from tqdm import tqdm

# Toggle on-off code text  
from IPython.display import HTML

HTML('''<script>
code_show=true; 
function code_toggle() {
 if (code_show){
 $('div.input').hide();
 } else {
 $('div.input').show();
 }
 code_show = !code_show
} 
$( document ).ready(code_toggle);
</script>
<form action="javascript:code_toggle()">
<input type="submit" value="Click to toggle on/off the raw code.">
</form>''')

### Function Definitions

In [27]:
from scipy.spatial.distance import euclidean

def nearest_k(query, objects, k, dist):
    """Return the indices to objects most similar to query
    
    Parameters
    ----------
    query : ndarray
        query object represented in the same form vector representation as the
        objects
    objects : ndarray
        vector-represented objects in the database; rows correspond to 
        objects, columns correspond to features
    k : int
        number of most similar objects to return
    dist : function
        accepts two ndarrays as parameters then returns their distance
    
    Returns
    -------
    ndarray
        Indices to the most similar objects in the database
    """
    return np.argsort([dist(query, o) for o in objects])[:k]

def inverse_transform(text_query):
    """Codes copied from data_building_and_cleaning. Credits to the 
    rightful owner."""
    
    import string
    punc = string.punctuation.replace('-', '').replace("'", '')
    table = str.maketrans(' ', ' ', punc)
    stripped = [w.translate(table) for w in [text_query.lower()]]

    ######################################################################

    # filter out stop words
    from nltk.corpus import stopwords

    stop_words = set(stopwords.words('english'))

    no_stop_words = []
    for word_list in stripped:
        word_list = word_list.split()
        words = [w for w in word_list if not w in stop_words]
        no_stop_words += [words]

    ######################################################################

    # nltk.download('wordnet')
    from nltk.stem import WordNetLemmatizer 

    lemmatizer = WordNetLemmatizer() 

    lematized = []
    for word_list in no_stop_words:
        words = [lemmatizer.lemmatize(w) for w in word_list]
        lematized += [words]

    title_cleaned = []
    for i in lematized:
        title_cleaned+=[" ".join(i)]

    return title_cleaned[0]

def IR_search(text_query, k, congress, criteria):
    from collections import defaultdict
    
    pres = {13:defaultdict(list), 14:defaultdict(list), 15:defaultdict(list), 
        16:defaultdict(list), 17:defaultdict(list), 18:defaultdict(list)}
    rec = {13:defaultdict(list), 14:defaultdict(list), 15:defaultdict(list), 
        16:defaultdict(list), 17:defaultdict(list), 18:defaultdict(list)}

    # congess filed_by scope filed_on primary_committee
    filtered = df_clean.query(f"congress == {congress} ").copy()

    tfidf = TfidfVectorizer(stop_words='english', min_df=0.001, max_df=0.999)
    tfidf.fit(filtered['combined_title'])
    X_idf  = tfidf.transform(filtered['combined_title']).toarray()
    X_feat_names = tfidf.get_feature_names()

    query = [inverse_transform(text_query)]                    
    print(f"Search term: {query[0]}")                          
    search_results = nearest_k(tfidf.transform(query).toarray()[0], X_idf, k, 
                               euclidean)
    pd.set_option('display.max_colwidth', None)
    print(f'Top {k} search results:\n\n')
    
    congress_ = list(range(13,19))
    
    for s in range(50):  # 50 is arbitrary, can be changed
        nearest = filtered.iloc[nearest_k(tfidf.transform(query).toarray()[0], 
                                          X_idf, 5, euclidean)]
        r = np.cumsum(nearest[criteria] == filtered[criteria].iloc[s])
        precision = r/np.arange(1, len(r)+1)
        n = sum(filtered[criteria] == filtered[criteria].iloc[s])
        recall = r/n
        recall = [0] + recall.tolist()
        precision = [1] + precision.tolist()
        pres[congress][s].append(precision)
        rec[congress][s].append(recall)

    ################################################################

    print(f'{text_query} precision_({criteria}): %.3f' 
          %np.mean([np.mean(pres[congress][key][0]) for key 
                    in pres[congress].keys()]))
    print(f'{text_query} recall_({criteria}): %.5f'
          %(np.mean([np.mean(rec[congress][key][0]) for key in 
                     rec[congress].keys()])))

    return filtered.iloc[search_results]

df_clean = pd.read_pickle('../data_cleaning/clusters_added.pkl')

bill_num = None
filed_by = None
scope = None
min_file_date = None
max_file_date = None
primary_committee = None

### 1st query: `minimum wage`

#### relevant criteria: `scope`

In [28]:
## sample querying

text_query = 'Minimum Wage'   ## type a sample query
k = 5                         ## specify no. of results
congress = 14                 ## specify which congress
criteria = 'scope'
IR_search(text_query, k, congress, criteria)

Search term: minimum wage
Top 5 search results:


Minimum Wage precision_(scope): 1.000
Minimum Wage recall_(scope): 0.00102


congress    type  bill_num  \
2783        14  Senate       103   
4256        14  Senate      1576   
4857        14  Senate      2177   
3259        14  Senate       579   
4073        14  Senate      1393   

                                                               title  \
2783                  INDIVIDUAL TAX EXEMPTION (MINIMUM WAGE EARNER)   
4256                       MINIMUM WAGE EARNERS INCOME TAX EXEMPTION   
4857                   INCOME TAX EXEMPTION FOR MINIMUM WAGE EARNERS   
3259  REGIONAL BOARD TO ISSUE WAGE ORDER ANNUALLY ON OR BEFORE MAY 1   
4073                                         LIVING WAGE ACT OF 2007   

       filed_on                         filed_by  \
2783 2007-06-30                     [Roxas, Mar]   
4256 2007-09-12                  [Villar, Manny]   
4857 2008-04-21        [Zubiri, Juan Miguel F, ]   
3259 2007-07-02  [Ejercito-Estrada, Jinggoy P, ]   
4073 2007-07-26             [Legarda, Loren B, ]   

                                                                                                                                                                                                                                            long_title  \
2783                             AN ACT EXEMPTING FROM INDIVIDUAL TAX WORKERS IN THE PRIVATE SECTOR RECEIVING MINIMUM WAGE, AS WELL AS GOVERNMENT WORKERS IN SALARY GRADES 1 TO 3, AMENDING FOR THIS PURPOSE REPUBLIC ACT 8424, AND FOR OTHER PURPOSES   
4256                                                   AN ACT EXEMPTING MINIMUM WAGE EARNERS FROM PAYMENT OF INCOME TAX AMENDING FOR THE PURPOSE RELEVANT PROVISIONS OF THE NATIONAL INTERNAL REVENUE CODE OF 1997, AS AMENDED, AND FOR OTHER PURPOSES   
4857                                                   AN ACT EXEMPTING MINIMUM WAGE EARNERS FROM PAYMENT OF INCOME TAX AMENDING FOR THE PURPOSE RELEVANT PROVISIONS OF THE NATIONAL INTERNAL REVENUE CODE OF 1997, AS AMENDED, AND FOR OTHER PURPOSES   
3259  AN ACT AMENDING ARTICLE 123 OF PRESIDENTIAL DECREE NO. 442, AS AMENDED, OTHERWISE KNOWN AS THE LABOR CODE OF THE PHILIPPINES, BY REQUIRING THE REGIONAL BOARD TO ISSUE A WAGE ORDER EVERY YEAR OR AN EXPLANATION FOR NOT ISSUING SUCH WAGE ORDER   
4073                                                      AN ACT DEFINING A LIVING WAGE FOR EMPLOYEES IN THE PUBLIC AND PRIVATE SECTORS, ESTABLISHING THE MECHANISM AND PROPER STANDARDS THEREFOR, APPROPRIATING FUNDS THEREFOR AND FOR OTHER PURPOSES   

         scope  \
2783  National   
4256  National   
4857  National   
3259  National   
4073  National   

                                                                                   subjects  \
2783  Individual Income Tax ExemptionMinimum Wage Earners (Individual Income Tax Exemption)   
4256                   Individual Income Tax ExemptionNational Internal Revenue Code (NIRC)   
4857                                     Individual Income Tax (Taxpayers Income Tax Rates)   
3259                                         Wage Order (Regional Board)Labor Code (Pd 442)   
4073                                                              Living WagePrivate Sector   

                                      primary_committee  \
2783                                     Ways and Means   
4256                                     Ways and Means   
4857                                     Ways and Means   
3259  Labor, Employment and Human Resources Development   
4073  Labor, Employment and Human Resources Development   

                                                                                                                                                                                                                         combined_title  \
2783                          individual tax exemption minimum wage earner act exempting individual tax worker private sector receiving minimum wage well government worker salary grade 1 3 amending purpose republic act 8424 purpose   
4256                                 

#### relevant criteria: `primary committee`

In [29]:
## sample querying

text_query = 'Minimum Wage'   ## type a sample query
k = 5                         ## specify no. of results
congress = 14                 ## specify which congress
criteria = 'primary_committee'
IR_search(text_query, k, congress, criteria)

Search term: minimum wage
Top 5 search results:


Minimum Wage precision_(primary_committee): 0.214
Minimum Wage recall_(primary_committee): 0.00202


congress    type  bill_num  \
2783        14  Senate       103   
4256        14  Senate      1576   
4857        14  Senate      2177   
3259        14  Senate       579   
4073        14  Senate      1393   

                                                               title  \
2783                  INDIVIDUAL TAX EXEMPTION (MINIMUM WAGE EARNER)   
4256                       MINIMUM WAGE EARNERS INCOME TAX EXEMPTION   
4857                   INCOME TAX EXEMPTION FOR MINIMUM WAGE EARNERS   
3259  REGIONAL BOARD TO ISSUE WAGE ORDER ANNUALLY ON OR BEFORE MAY 1   
4073                                         LIVING WAGE ACT OF 2007   

       filed_on                         filed_by  \
2783 2007-06-30                     [Roxas, Mar]   
4256 2007-09-12                  [Villar, Manny]   
4857 2008-04-21        [Zubiri, Juan Miguel F, ]   
3259 2007-07-02  [Ejercito-Estrada, Jinggoy P, ]   
4073 2007-07-26             [Legarda, Loren B, ]   

                                                                                                                                                                                                                                            long_title  \
2783                             AN ACT EXEMPTING FROM INDIVIDUAL TAX WORKERS IN THE PRIVATE SECTOR RECEIVING MINIMUM WAGE, AS WELL AS GOVERNMENT WORKERS IN SALARY GRADES 1 TO 3, AMENDING FOR THIS PURPOSE REPUBLIC ACT 8424, AND FOR OTHER PURPOSES   
4256                                                   AN ACT EXEMPTING MINIMUM WAGE EARNERS FROM PAYMENT OF INCOME TAX AMENDING FOR THE PURPOSE RELEVANT PROVISIONS OF THE NATIONAL INTERNAL REVENUE CODE OF 1997, AS AMENDED, AND FOR OTHER PURPOSES   
4857                                                   AN ACT EXEMPTING MINIMUM WAGE EARNERS FROM PAYMENT OF INCOME TAX AMENDING FOR THE PURPOSE RELEVANT PROVISIONS OF THE NATIONAL INTERNAL REVENUE CODE OF 1997, AS AMENDED, AND FOR OTHER PURPOSES   
3259  AN ACT AMENDING ARTICLE 123 OF PRESIDENTIAL DECREE NO. 442, AS AMENDED, OTHERWISE KNOWN AS THE LABOR CODE OF THE PHILIPPINES, BY REQUIRING THE REGIONAL BOARD TO ISSUE A WAGE ORDER EVERY YEAR OR AN EXPLANATION FOR NOT ISSUING SUCH WAGE ORDER   
4073                                                      AN ACT DEFINING A LIVING WAGE FOR EMPLOYEES IN THE PUBLIC AND PRIVATE SECTORS, ESTABLISHING THE MECHANISM AND PROPER STANDARDS THEREFOR, APPROPRIATING FUNDS THEREFOR AND FOR OTHER PURPOSES   

         scope  \
2783  National   
4256  National   
4857  National   
3259  National   
4073  National   

                                                                                   subjects  \
2783  Individual Income Tax ExemptionMinimum Wage Earners (Individual Income Tax Exemption)   
4256                   Individual Income Tax ExemptionNational Internal Revenue Code (NIRC)   
4857                                     Individual Income Tax (Taxpayers Income Tax Rates)   
3259                                         Wage Order (Regional Board)Labor Code (Pd 442)   
4073                                                              Living WagePrivate Sector   

                                      primary_committee  \
2783                                     Ways and Means   
4256                                     Ways and Means   
4857                                     Ways and Means   
3259  Labor, Employment and Human Resources Development   
4073  Labor, Employment and Human Resources Development   

                                                                                                                                                                                                                         combined_title  \
2783                          individual tax exemption minimum wage earner act exempting individual tax worker private sector receiving minimum wage well government worker salary grade 1 3 amending purpose republic act 8424 purpose   
4256                                 

### 2nd query: `Overseas Filipino Workers`

#### relevant criteria: `scope`

In [30]:
## sample querying

text_query = 'Overseas Filipino Workers'   ## type a sample query
k = 5                         ## specify no. of results
congress = 14                 ## specify which congress
criteria = 'scope'
IR_search(text_query, k, congress, criteria)

Search term: overseas filipino worker
Top 5 search results:


Overseas Filipino Workers precision_(scope): 1.000
Overseas Filipino Workers recall_(scope): 0.00102


congress    type  bill_num  \
5008        14  Senate      2328   
2834        14  Senate       154   
2734        14  Senate        54   
5378        14  Senate      2698   
5597        14  Senate      2917   

                                                                         title  \
5008                        OVERSEAS FILIPINO WORKERS PENSION FUND ACT OF 2008   
2834                REVISED MIGRANT WORKERS AND OVERSEAS FILIPINOS ACT OF 2007   
2734                        MIGRANT WORKERS AND OVERSEAS FILIPINOS, AMENDMENTS   
5378  MANDATORY EMPLOYERS' INSURANCE FOR OVERSEAS FILIPINO WORKERS ACT OF 2008   
5597                                                 OVERSEAS FILIPINO WORKERS   

       filed_on                             filed_by  \
5008 2008-05-27  [Trillanes IV, Antonio "Sonny" F, ]   
2834 2007-06-30      [Ejercito-Estrada, Jinggoy P, ]   
2734 2007-06-30           [Lapid, Manuel "Lito" M, ]   
5378 2008-11-06                      [Villar, Manny]   
5597 2008-11-27                      [Villar, Manny]   

                                                                                                                                                                                                                                                                                                    long_title  \
5008                                                                                                                                                                                                                    AN ACT CREATING THE OVERSEAS FILIPINO WORKERS PENSION AND APPROPRIATING FUNDS THEREFOR   
2834         AN ACT AMENDING REPUBLIC ACT NO. 8042, OTHERWISE KNOWN AS THE MIGRANT WORKERS AND OVERSEAS FILIPINOS ACT OF 1995, FURTHER IMPROVING THE HIGHER STANDARDS OF PROTECTION AND PROMOTION OF THE WELFARE OF MIGRANT WORKERS, THEIR FAMILIES AND OVERSEAS FILIPINOS IN DISTRESS, AND FOR OTHER PURPOSES   
2734  AN ACT PROVIDING RETIREMENT, INSURANCE, HEALTH AND EDUCATIONAL BENEFITS TO OVERSEAS CONTRACT WORKERS (OCWS) AND THEIR BENEFICIARIES AMENDING FOR THE PURPOSE CERTAIN PROVISIONS OF REPUBLIC ACT NO. 8042 OTHERWISE KNOWN AS ‘MIGRANT WORKERS AND OVERSEAS FILIPINOS ACT OF 1995’, AND FOR OTHER PURPOSES   
5378                                                                                                                                              AN ACT TO PROVIDE A MANDATORY EMPLOYERS' INSURANCE FOR ALL  LEGALLY EMPLOYED OVERSEAS FILIPINO WORKERS, PROVIDING MECHANISMS THEREFOR AND FOR OTHER PURPOSES   
5597                                                                                                                                                                                               AN ACT GRANTING BENEFITS AND SPECIAL PRIVILEGES TO FORMER OVERSEAS FILIPINO WORKERS, AND FOR OTHER PURPOSES   

         scope  \
5008  National   
2834  National   
2734  National   
5378  National   
5597  National   

                                                                                                                           subjects  \
5008                                                              Overseas Contract Workers (OCWS)/Overseas Filipino Workers (OFWs)   
2834    Revised Migrant Workers and Overseas Filipinos Act of 2007Overseas Contract Workers (OCWS)/Overseas Filipino Workers (OFWs)   
2734  Overseas Contract Workers (OCWS)/Overseas Filipino Workers (OFWs)Migrant Workers and Overseas Filipinos (RA 8042, Amendments)   
5378                                                              Overseas Contract Workers (OCWS)/Overseas Filipino Workers (OFWs)   
5597                                                              Overseas Contract Workers (OCWS)/Overseas Filipino Workers (OFWs)   

                                      primary_committee  \
5008  Labor, Employment and Human Resources Development   
2834  Labor, Employment and Human Resources Development   
2734  Labor, Employment a

#### relevant criteria: `primary_committee`

In [31]:
## sample querying

text_query = 'Overseas Filipino Workers'   ## type a sample query
k = 5                         ## specify no. of results
congress = 14                 ## specify which congress
criteria = 'primary_committee'
IR_search(text_query, k, congress, criteria)

Search term: overseas filipino worker
Top 5 search results:


Overseas Filipino Workers precision_(primary_committee): 0.200
Overseas Filipino Workers recall_(primary_committee): 0.00071


congress    type  bill_num  \
5008        14  Senate      2328   
2834        14  Senate       154   
2734        14  Senate        54   
5378        14  Senate      2698   
5597        14  Senate      2917   

                                                                         title  \
5008                        OVERSEAS FILIPINO WORKERS PENSION FUND ACT OF 2008   
2834                REVISED MIGRANT WORKERS AND OVERSEAS FILIPINOS ACT OF 2007   
2734                        MIGRANT WORKERS AND OVERSEAS FILIPINOS, AMENDMENTS   
5378  MANDATORY EMPLOYERS' INSURANCE FOR OVERSEAS FILIPINO WORKERS ACT OF 2008   
5597                                                 OVERSEAS FILIPINO WORKERS   

       filed_on                             filed_by  \
5008 2008-05-27  [Trillanes IV, Antonio "Sonny" F, ]   
2834 2007-06-30      [Ejercito-Estrada, Jinggoy P, ]   
2734 2007-06-30           [Lapid, Manuel "Lito" M, ]   
5378 2008-11-06                      [Villar, Manny]   
5597 2008-11-27                      [Villar, Manny]   

                                                                                                                                                                                                                                                                                                    long_title  \
5008                                                                                                                                                                                                                    AN ACT CREATING THE OVERSEAS FILIPINO WORKERS PENSION AND APPROPRIATING FUNDS THEREFOR   
2834         AN ACT AMENDING REPUBLIC ACT NO. 8042, OTHERWISE KNOWN AS THE MIGRANT WORKERS AND OVERSEAS FILIPINOS ACT OF 1995, FURTHER IMPROVING THE HIGHER STANDARDS OF PROTECTION AND PROMOTION OF THE WELFARE OF MIGRANT WORKERS, THEIR FAMILIES AND OVERSEAS FILIPINOS IN DISTRESS, AND FOR OTHER PURPOSES   
2734  AN ACT PROVIDING RETIREMENT, INSURANCE, HEALTH AND EDUCATIONAL BENEFITS TO OVERSEAS CONTRACT WORKERS (OCWS) AND THEIR BENEFICIARIES AMENDING FOR THE PURPOSE CERTAIN PROVISIONS OF REPUBLIC ACT NO. 8042 OTHERWISE KNOWN AS ‘MIGRANT WORKERS AND OVERSEAS FILIPINOS ACT OF 1995’, AND FOR OTHER PURPOSES   
5378                                                                                                                                              AN ACT TO PROVIDE A MANDATORY EMPLOYERS' INSURANCE FOR ALL  LEGALLY EMPLOYED OVERSEAS FILIPINO WORKERS, PROVIDING MECHANISMS THEREFOR AND FOR OTHER PURPOSES   
5597                                                                                                                                                                                               AN ACT GRANTING BENEFITS AND SPECIAL PRIVILEGES TO FORMER OVERSEAS FILIPINO WORKERS, AND FOR OTHER PURPOSES   

         scope  \
5008  National   
2834  National   
2734  National   
5378  National   
5597  National   

                                                                                                                           subjects  \
5008                                                              Overseas Contract Workers (OCWS)/Overseas Filipino Workers (OFWs)   
2834    Revised Migrant Workers and Overseas Filipinos Act of 2007Overseas Contract Workers (OCWS)/Overseas Filipino Workers (OFWs)   
2734  Overseas Contract Workers (OCWS)/Overseas Filipino Workers (OFWs)Migrant Workers and Overseas Filipinos (RA 8042, Amendments)   
5378                                                              Overseas Contract Workers (OCWS)/Overseas Filipino Workers (OFWs)   
5597                                                              Overseas Contract Workers (OCWS)/Overseas Filipino Workers (OFWs)   

                                      primary_committee  \
5008  Labor, Employment and Human Resources Development   
2834  Labor, Employment and Human Resources Development   
2734  Labor, Employment a

### 3rd query: `Illegal Drugs`

#### relevant criteria: `scope`

In [32]:
## sample querying

text_query = 'Illegal Drugs'   ## type a sample query
k = 5                         ## specify no. of results
congress = 14                 ## specify which congress
criteria = 'scope'
IR_search(text_query, k, congress, criteria)

Search term: illegal drug
Top 5 search results:


Illegal Drugs precision_(scope): 1.000
Illegal Drugs recall_(scope): 0.00102


congress    type  bill_num  \
3964        14  Senate      1284   
3996        14  Senate      1316   
4430        14  Senate      1750   
2733        14  Senate        53   
3998        14  Senate      1318   

                                                                  title  \
3964                                             DRUGS DEMAND REDUCTION   
3996                                          DRUG DEALER LIABILITY ACT   
4430                                      DRUG-FREE BORDERS ACT OF 2007   
2733                                              ILLEGAL NUMBERS GAMES   
3998  THE DRUG ABUSE PREVENTION PROGRAM IN EDUCATIONAL INSTITUTIONS ACT   

       filed_on                     filed_by  \
3964 2007-07-18  [Defensor Santiago, Miriam]   
3996 2007-07-24  [Defensor Santiago, Miriam]   
4430 2007-10-16  [Defensor Santiago, Miriam]   
2733 2007-06-30   [Lapid, Manuel "Lito" M, ]   
3998 2007-07-24  [Defensor Santiago, Miriam]   

                                                                                                                                                                                                                                                                                           long_title  \
3964                                                                                  AN ACT PROVIDING FOR PROGRAMS TO FACILITATE A SIGNIFICANT REDUCTION IN THE INCIDENCE AND PREVALENCE OF SUBSTANCE ABUSE THROUGH REDUCING THE DEMAND FOR ILLEGAL DRUGS AND THE INAPPROPRIATE USE OF ILLEGAL DRUGS   
3996                                                                                                                                                                                            AN ACT PROVIDING FOR THE CIVIL LIABILITY OF DRUG DEALERS IN FAVOR OF PERSONS INJURED BY ILLEGAL DRUGS   
4430                                                                                                                                                              AN ACT SECURING THE PHILIPPINES' BORDERS TO MAKE MORE EFFECTIVE THE PROHIBITION AGAINST THE ENTRY OF ILLEGAL DRUGS INTO THE COUNTRY   
2733  AN ACT AMENDING CERTAIN PROVISIONS OF REPUBLIC ACT NO. 9287 OTHERWISE KNOWN AS ‘AN ACT INCREASING THE PENALTIES OF ILLEGAL NUMBERS GAMES’, BY INCLUDING ALTOGETHER ALL FORMS OF ILLEGAL GAMBLING, CREATING THE ANTI-ILLEGAL GAMBLING BOARD, APPROPRIATING FUNDS THEREFOR AND FOR OTHER PURPOSES   
3998                                                                                                                    AN ACT MANDATING THE ESTABLISHMENT OF A PROGRAM OF EDUCATION DESIGNED TO HELP ALL MEMBERS OF THE COMMUNITY OF AN EDUCATIONAL INSTITUTION AVOID INVOLVEMENT WITH ILLEGAL DRUGS   

         scope  \
3964  National   
3996  National   
4430  National   
2733  National   
3998  National   

                                                                        subjects  \
3964                                          Drug Demand ReductionIllegal Drugs   
3996                                          Drug Dealer LiabilityIllegal Drugs   
4430                                                           Drug-Free Borders   
2733        Illegal Number Games (RA 9287)Anti-Illegal Gambling Board (Creation)   
3998  Drug Abuse Prevention Program in Educational Institutions ActIllegal Drugs   

                      primary_committee  \
3964   Public Order and Dangerous Drugs   
3996   Public Order and Dangerous Drugs   
4430   Public Order and Dangerous Drugs   
2733                Games and Amusement   
3998  Basic Education, Arts and Culture   

                                                                                                                                                                                                                             combined_title  \
3964                                                          drug demand reduction act providing program facilitate significant reduction incidence preva

#### relevant criteria: `primary_committee`

In [33]:
## sample querying

text_query = 'Illegal Drugs'   ## type a sample query
k = 5                         ## specify no. of results
congress = 14                 ## specify which congress
criteria = 'primary_committee'
IR_search(text_query, k, congress, criteria)

Search term: illegal drug
Top 5 search results:


Illegal Drugs precision_(primary_committee): 0.240
Illegal Drugs recall_(primary_committee): 0.00231


congress    type  bill_num  \
3964        14  Senate      1284   
3996        14  Senate      1316   
4430        14  Senate      1750   
2733        14  Senate        53   
3998        14  Senate      1318   

                                                                  title  \
3964                                             DRUGS DEMAND REDUCTION   
3996                                          DRUG DEALER LIABILITY ACT   
4430                                      DRUG-FREE BORDERS ACT OF 2007   
2733                                              ILLEGAL NUMBERS GAMES   
3998  THE DRUG ABUSE PREVENTION PROGRAM IN EDUCATIONAL INSTITUTIONS ACT   

       filed_on                     filed_by  \
3964 2007-07-18  [Defensor Santiago, Miriam]   
3996 2007-07-24  [Defensor Santiago, Miriam]   
4430 2007-10-16  [Defensor Santiago, Miriam]   
2733 2007-06-30   [Lapid, Manuel "Lito" M, ]   
3998 2007-07-24  [Defensor Santiago, Miriam]   

                                                                                                                                                                                                                                                                                           long_title  \
3964                                                                                  AN ACT PROVIDING FOR PROGRAMS TO FACILITATE A SIGNIFICANT REDUCTION IN THE INCIDENCE AND PREVALENCE OF SUBSTANCE ABUSE THROUGH REDUCING THE DEMAND FOR ILLEGAL DRUGS AND THE INAPPROPRIATE USE OF ILLEGAL DRUGS   
3996                                                                                                                                                                                            AN ACT PROVIDING FOR THE CIVIL LIABILITY OF DRUG DEALERS IN FAVOR OF PERSONS INJURED BY ILLEGAL DRUGS   
4430                                                                                                                                                              AN ACT SECURING THE PHILIPPINES' BORDERS TO MAKE MORE EFFECTIVE THE PROHIBITION AGAINST THE ENTRY OF ILLEGAL DRUGS INTO THE COUNTRY   
2733  AN ACT AMENDING CERTAIN PROVISIONS OF REPUBLIC ACT NO. 9287 OTHERWISE KNOWN AS ‘AN ACT INCREASING THE PENALTIES OF ILLEGAL NUMBERS GAMES’, BY INCLUDING ALTOGETHER ALL FORMS OF ILLEGAL GAMBLING, CREATING THE ANTI-ILLEGAL GAMBLING BOARD, APPROPRIATING FUNDS THEREFOR AND FOR OTHER PURPOSES   
3998                                                                                                                    AN ACT MANDATING THE ESTABLISHMENT OF A PROGRAM OF EDUCATION DESIGNED TO HELP ALL MEMBERS OF THE COMMUNITY OF AN EDUCATIONAL INSTITUTION AVOID INVOLVEMENT WITH ILLEGAL DRUGS   

         scope  \
3964  National   
3996  National   
4430  National   
2733  National   
3998  National   

                                                                        subjects  \
3964                                          Drug Demand ReductionIllegal Drugs   
3996                                          Drug Dealer LiabilityIllegal Drugs   
4430                                                           Drug-Free Borders   
2733        Illegal Number Games (RA 9287)Anti-Illegal Gambling Board (Creation)   
3998  Drug Abuse Prevention Program in Educational Institutions ActIllegal Drugs   

                      primary_committee  \
3964   Public Order and Dangerous Drugs   
3996   Public Order and Dangerous Drugs   
4430   Public Order and Dangerous Drugs   
2733                Games and Amusement   
3998  Basic Education, Arts and Culture   

                                                                                                                                                                                                                             combined_title  \
3964                                                          drug demand reduction act providing program facilitate significant reduction incidence preva

### 4th query: `Death Penalty`

#### relevant criteria: `scope`

In [34]:
## sample querying

text_query = 'Death Penalty'   ## type a sample query
k = 5                         ## specify no. of results
congress = 14                 ## specify which congress
criteria = 'scope'
IR_search(text_query, k, congress, criteria)

Search term: death penalty
Top 5 search results:


Death Penalty precision_(scope): 1.000
Death Penalty recall_(scope): 0.00102


congress    type  bill_num  \
5002        14  Senate      2322   
5665        14  Senate      2985   
3105        14  Senate       425   
4873        14  Senate      2193   
2966        14  Senate       286   

                                                                                               title  \
5002                                              REIMPOSING DEATH PENALTY ON CERTAIN HEINOUS CRIMES   
5665                                                               TOY-INFLICTED INJURIES AND DEATHS   
3105                                                        DEATH BENEFITS FOR SEAFARERS ACT OF 2007   
4873                                                                              ANTI-SALVAGING LAW   
2966  SUSPENDING PAYMENT OF ESTATE TAX ON FAMILY HOME BY SURVIVING SPOUSE IN CASE OF DEATH OF SPOUSE   

       filed_on                             filed_by  \
5002 2008-05-26            [Zubiri, Juan Miguel F, ]   
5665 2009-01-15          [Defensor Santiago, Miriam]   
3105 2007-07-02      [Ejercito-Estrada, Jinggoy P, ]   
4873 2008-04-23  [Trillanes IV, Antonio "Sonny" F, ]   
2966 2007-06-30         [Gordon, Richard "Dick" J, ]   

                                                                                                                                                                                                                                                                               long_title  \
5002                                                                                                            AN ACT REIMPOSING THE PENALTY OF DEATH ON CERTAIN HEINOUS CRIMES, AMENDING FOR THE PURPOSE CERTAIN SECTIONS OF THE REVISED PENAL CODE, AS AMENDED, AND FOR OTHER PURPOSES   
5665                                                                                                                                                                                                                                  AN ACT TO PREVENT TOY-INFLICTED INJURIES AND DEATHS   
3105  AN ACT FIXING THE MINIMUM STANDARD FOR DEATH BENEFITS TO FILIPINO SEAFARERS EMPLOYED OVERSEAS AND ESTABLISHING A MECHANISM AT THE PHILIPPINE OVERSEAS EMPLOYMENT ADMINISTRATION FOR ITS REGULAR REVIEW AND REALIGNMENT TO PREVAILING INTERNATIONAL STANDARDS AND FOR OTHER PURPOSES   
4873                                                                     AN ACT QUALIFYING SALVAGING OR EXTRAJUDICIAL KILLING BY ANY PUBLIC OFFICER, PERSON IN AUTHORITY OR AGENT OF A PERSON IN AUTHORITY AS A HEINOUS CRIME, IMPOSING THE DEATH PENALTY THEREFOR AND FOR OTHER PURPOSES   
2966                                                                                                                                                AN ACT AMENDING SECTION 86(A) (4) OF REPUBLIC ACT NO. 8424, AS AMENDED, OTHERWISE KNOWN AS THE NATIONAL INTERNAL REVENUE CODE OF 1997   

         scope  \
5002  National   
5665  National   
3105  National   
4873  National   
2966  National   

                                                                                               subjects  \
5002                                Revised Penal Code (RA 3815; Amendments)Death PenaltyHeinous Crimes   
5665                                                                                               Toys   
3105                                      SeafarersPhilippine Overseas Employment Administration (POEA)   
4873                                                                                     Heinous Crimes   
2966  National Internal Revenue Code (NIRC)Family HomeR.A.No. 8424 (Tax Reform Act of 1997), Amendments   

                                      primary_committee  \
5002                           Justice and Human Rights   
5665               Trade, Commerce and Entrepreneurship   
3105  Labor, Employment and Human Resources Development   
4873                           Justice and Human Rights   
2966                                     Ways and Means   

         

#### relevant criteria: `primary_committee`

In [35]:
## sample querying

text_query = 'Death Penalty'   ## type a sample query
k = 5                         ## specify no. of results
congress = 14                 ## specify which congress
criteria = 'primary_committee'
IR_search(text_query, k, congress, criteria)

Search term: death penalty
Top 5 search results:


Death Penalty precision_(primary_committee): 0.201
Death Penalty recall_(primary_committee): 0.00048


congress    type  bill_num  \
5002        14  Senate      2322   
5665        14  Senate      2985   
3105        14  Senate       425   
4873        14  Senate      2193   
2966        14  Senate       286   

                                                                                               title  \
5002                                              REIMPOSING DEATH PENALTY ON CERTAIN HEINOUS CRIMES   
5665                                                               TOY-INFLICTED INJURIES AND DEATHS   
3105                                                        DEATH BENEFITS FOR SEAFARERS ACT OF 2007   
4873                                                                              ANTI-SALVAGING LAW   
2966  SUSPENDING PAYMENT OF ESTATE TAX ON FAMILY HOME BY SURVIVING SPOUSE IN CASE OF DEATH OF SPOUSE   

       filed_on                             filed_by  \
5002 2008-05-26            [Zubiri, Juan Miguel F, ]   
5665 2009-01-15          [Defensor Santiago, Miriam]   
3105 2007-07-02      [Ejercito-Estrada, Jinggoy P, ]   
4873 2008-04-23  [Trillanes IV, Antonio "Sonny" F, ]   
2966 2007-06-30         [Gordon, Richard "Dick" J, ]   

                                                                                                                                                                                                                                                                               long_title  \
5002                                                                                                            AN ACT REIMPOSING THE PENALTY OF DEATH ON CERTAIN HEINOUS CRIMES, AMENDING FOR THE PURPOSE CERTAIN SECTIONS OF THE REVISED PENAL CODE, AS AMENDED, AND FOR OTHER PURPOSES   
5665                                                                                                                                                                                                                                  AN ACT TO PREVENT TOY-INFLICTED INJURIES AND DEATHS   
3105  AN ACT FIXING THE MINIMUM STANDARD FOR DEATH BENEFITS TO FILIPINO SEAFARERS EMPLOYED OVERSEAS AND ESTABLISHING A MECHANISM AT THE PHILIPPINE OVERSEAS EMPLOYMENT ADMINISTRATION FOR ITS REGULAR REVIEW AND REALIGNMENT TO PREVAILING INTERNATIONAL STANDARDS AND FOR OTHER PURPOSES   
4873                                                                     AN ACT QUALIFYING SALVAGING OR EXTRAJUDICIAL KILLING BY ANY PUBLIC OFFICER, PERSON IN AUTHORITY OR AGENT OF A PERSON IN AUTHORITY AS A HEINOUS CRIME, IMPOSING THE DEATH PENALTY THEREFOR AND FOR OTHER PURPOSES   
2966                                                                                                                                                AN ACT AMENDING SECTION 86(A) (4) OF REPUBLIC ACT NO. 8424, AS AMENDED, OTHERWISE KNOWN AS THE NATIONAL INTERNAL REVENUE CODE OF 1997   

         scope  \
5002  National   
5665  National   
3105  National   
4873  National   
2966  National   

                                                                                               subjects  \
5002                                Revised Penal Code (RA 3815; Amendments)Death PenaltyHeinous Crimes   
5665                                                                                               Toys   
3105                                      SeafarersPhilippine Overseas Employment Administration (POEA)   
4873                                                                                     Heinous Crimes   
2966  National Internal Revenue Code (NIRC)Family HomeR.A.No. 8424 (Tax Reform Act of 1997), Amendments   

                                      primary_committee  \
5002                           Justice and Human Rights   
5665               Trade, Commerce and Entrepreneurship   
3105  Labor, Employment and Human Resources Development   
4873                           Justice and Human Rights   
2966                                     Ways and Means   

         

### 5th query: `Climate Change`

#### relevant criteria: `scope`

In [36]:
## sample querying

text_query = 'Climate Change'   ## type a sample query
k = 5                         ## specify no. of results
congress = 14                 ## specify which congress
criteria = 'scope'
IR_search(text_query, k, congress, criteria)

Search term: climate change
Top 5 search results:


Climate Change precision_(scope): 1.000
Climate Change recall_(scope): 0.00102


congress    type  bill_num                                   title  \
5121        14  Senate      2441    CLIMATE CHANGE EDUCATION ACT OF 2008   
5068        14  Senate      2388  CLIMATE CHANGE EDUCATIONAL PROGRAM ACT   
5039        14  Senate      2359  DRINKING WATER ADAPTATION RESEARCH ACT   
5016        14  Senate      2336    NATIONAL CLIMATE PROGRAM ACT OF 2008   
3935        14  Senate      1255                        EXACT CHANGE ACT   

       filed_on                     filed_by  \
5121 2008-07-14   [Lapid, Manuel "Lito" M, ]   
5068 2008-06-10  [Defensor Santiago, Miriam]   
5039 2008-06-02  [Defensor Santiago, Miriam]   
5016 2008-05-28  [Defensor Santiago, Miriam]   
3935 2007-07-18  [Defensor Santiago, Miriam]   

                                                                                               long_title  \
5121  AN ACT INSTITUTIONALIZING THE CLIMATE CHANGE EDUCATION AND AWARENESS PROGRAM AND FOR OTHER PURPOSES   
5068                                           AN ACT ESTABLISHING THE CLIMATE CHANGE EDUCATIONAL PROGRAM   
5039                                                CLIMATE CHANGE DRINKING WATER ADAPTATION RESEARCH ACT   
5016                                                     AN ACT ESTABLISHING THE NATIONAL CLIMATE PROGRAM   
3935                           AN ACT REQUIRING BUSINESS ESTABLISHMENTS TO GIVE EXACT CHANGE TO CONSUMERS   

         scope                      subjects  \
5121  National                Climate Change   
5068  National                Climate Change   
5039  National  Climate ChangeDrinking Water   
5016  National                Climate Change   
3935  National     Business/Entrepreneurship   

                                      primary_committee  \
5121                  Basic Education, Arts and Culture   
5068  Environment, Natural Resources and Climate Change   
5039  Environment, Natural Resources and Climate Change   
5016  Environment, Natural Resources and Climate Change   
3935               Trade, Commerce and Entrepreneurship   

                                                                                                   combined_title  \
5121  climate change education act 2008 act institutionalizing climate change education awareness program purpose   
5068                   climate change educational program act act establishing climate change educational program   
5039                                                        climate change drinking water adaptation research act   
5016                                  national climate program act 2008 act establishing national climate program   
3935                             exact change act act requiring business establishment give exact change consumer   

      clustering_ward_a  clustering_ward_b  clustering_ward_c  
5121                  0                  2                  4  
5068                  0                  2                  4  
5039                  0                  2                  4  
5016                  0                  1                 24  
3935                  0                  2                  4

#### relevant criteria: `primary_committee`

In [37]:
## sample querying

text_query = 'Climate Change'   ## type a sample query
k = 5                         ## specify no. of results
congress = 14                 ## specify which congress
criteria = 'primary_committee'
IR_search(text_query, k, congress, criteria)

Search term: climate change
Top 5 search results:


Climate Change precision_(primary_committee): 0.199
Climate Change recall_(primary_committee): 0.00067


congress    type  bill_num                                   title  \
5121        14  Senate      2441    CLIMATE CHANGE EDUCATION ACT OF 2008   
5068        14  Senate      2388  CLIMATE CHANGE EDUCATIONAL PROGRAM ACT   
5039        14  Senate      2359  DRINKING WATER ADAPTATION RESEARCH ACT   
5016        14  Senate      2336    NATIONAL CLIMATE PROGRAM ACT OF 2008   
3935        14  Senate      1255                        EXACT CHANGE ACT   

       filed_on                     filed_by  \
5121 2008-07-14   [Lapid, Manuel "Lito" M, ]   
5068 2008-06-10  [Defensor Santiago, Miriam]   
5039 2008-06-02  [Defensor Santiago, Miriam]   
5016 2008-05-28  [Defensor Santiago, Miriam]   
3935 2007-07-18  [Defensor Santiago, Miriam]   

                                                                                               long_title  \
5121  AN ACT INSTITUTIONALIZING THE CLIMATE CHANGE EDUCATION AND AWARENESS PROGRAM AND FOR OTHER PURPOSES   
5068                                           AN ACT ESTABLISHING THE CLIMATE CHANGE EDUCATIONAL PROGRAM   
5039                                                CLIMATE CHANGE DRINKING WATER ADAPTATION RESEARCH ACT   
5016                                                     AN ACT ESTABLISHING THE NATIONAL CLIMATE PROGRAM   
3935                           AN ACT REQUIRING BUSINESS ESTABLISHMENTS TO GIVE EXACT CHANGE TO CONSUMERS   

         scope                      subjects  \
5121  National                Climate Change   
5068  National                Climate Change   
5039  National  Climate ChangeDrinking Water   
5016  National                Climate Change   
3935  National     Business/Entrepreneurship   

                                      primary_committee  \
5121                  Basic Education, Arts and Culture   
5068  Environment, Natural Resources and Climate Change   
5039  Environment, Natural Resources and Climate Change   
5016  Environment, Natural Resources and Climate Change   
3935               Trade, Commerce and Entrepreneurship   

                                                                                                   combined_title  \
5121  climate change education act 2008 act institutionalizing climate change education awareness program purpose   
5068                   climate change educational program act act establishing climate change educational program   
5039                                                        climate change drinking water adaptation research act   
5016                                  national climate program act 2008 act establishing national climate program   
3935                             exact change act act requiring business establishment give exact change consumer   

      clustering_ward_a  clustering_ward_b  clustering_ward_c  
5121                  0                  2                  4  
5068                  0                  2                  4  
5039                  0                  2                  4  
5016                  0                  1                 24  
3935                  0                  2                  4

### Summary

**Summary of Precision Values from Manual and Automated**<br>

|query|No. of Results|No. of Relevant Results(Manual)|Precision (Manual)|Precision_Scope (IR)|Precision_PrimaryCommittee (IR)|
|-----|--------------|-------------------------------|------------------|--------------------|-------------------------------|
|`minimum wage`| 5 | 5 | 1.00 | 1.00 | 0.214 |
`overseas filipino workers`| 5 | 5 | 1.00 | 1.00 | 0.200 |
`illegal drugs`| 5 | 4 | 0.80 | 1.00 |  0.240 |
`death penalty`| 5 | 2 | 0.40 | 1.00 | 0.201 |
`climate change`| 5 | 4 | 0.80 | 1.00 | 0.199 |